sympy/solvers/solvers.py

```diff
@@ -1311,45 +1354,45 @@ def _solve(f, *symbols, **flags):
 
     not_impl_msg = "No algorithms are implemented to solve equation %s"
 
-    if len(symbols) != 1:
+    if len(symbols) != 1:                            # solve(a*x + 2*x + b - c, a, b)
         soln = None
-        free = f.free_symbols
-        ex = free - set(symbols)
+        free = f.free_symbols                        # {a, b, c, x}
+        ex = free - set(symbols)                     # {c, x}
         if len(ex) != 1:
-            ind, dep = f.as_independent(*symbols)
-            ex = ind.free_symbols & dep.free_symbols
+            ind, dep = f.as_independent(*symbols)    # (2*x - c, a*x + b)
+            ex = ind.free_symbols & dep.free_symbols # {x, c} & {a, x, b} -> {x}
         if len(ex) == 1:
```

Comment:
Although `solve_undetermined_coeffs((a+b)*x + b-c, (a,b), x) -> {a:-c, b:c}` I don't think this should be added here -- that seems to be an odd type of coefficient matching. But if we did it would look like:
```python
if len(ex) > 1:
    ex = dep.free_symbols - set(symbols)
if len(ex) == 1:
...
```
Requesting `(a, c)`  would make sense, however, without making this change.